xbps-src: overwrite packages in local repo

Chocimier opened the pull request:
A better default than wasting cycles and confusing people why xbps-install -f didn't work.
Builder needs non-default behavior.

@void-linux/pkg-committers

ericonr commented on common/xbps-src/shutils/pkgtarget.sh:
```diff
@@ -1,5 +1,24 @@
+    if [ -z "$XBPS_PRESERVE_PKGS" ]; then
+        return
+    fi
+    if [ -n "$XBPS_TARGET_MACHINE" ]; then
+        arch=$XBPS_TARGET_MACHINE
+    else
+        arch=$XBPS_MACHINE
+    fi
+    if [ -z "$XBPS_CROSS_BUILD" -a -n "$XBPS_ARCH" -a "$XBPS_ARCH" != "$XBPS_TARGET_MACHINE" ]; then
+        arch=${XBPS_ARCH}
+    fi
```
Shouldn't `XBPS_TARGET_MACHINE` always be set? We use it in templates as though it's always set...

ericonr reviewed (commented):
It definitely makes sense. I suppose we could deploy the configuration to builders with the var, and once that's done, merge this PR.

General notes:

- no command line flag to change the behavior; this would be the opposite of `-f`; in the same vein, no way to override `XBPS_PRESERVE_PKGS` with the existing `-f`, which might be useful
- this changes the tool behavior, not sure I'm comfortable with that; we should definitely document the full extent of the change, at least. At the moment, an existing package will still allow a build, it just won't move the existing packages to hostdir and overwrite them in repodata; if there suddenly appears a new subpackage or whatever, that will still be added to repodata (this might be desired behavior sometimes, I know I have used it this way). `XBPS_PRESERVE_PKGS` will, instead, not run any *builds* for existing packages, which is quite different from the current default.

Chocimier commented:
Added `-f`, simplified `XBPS_TARGET_MACHINE`.

As for unsetting XBPS_PRESERVE_PKGS with flag, is this really more useful than unsetting XBPS_CHECK_PKGS? Do we want parity  between flags and variables in general?

For documentation, anything more needed?

> if there suddenly appears a new subpackage

Well, as proposed, xbps-src only looks whether package from command line argument is there, and if not, it overrides all other packages build together. IMO good enough and better than as is, i.e. preserve existing packages but index new, possibly with conflicting files.

Chocimier commented:
@Duncaen, fixed

Chocimier commented:
@beanpole135, @RossComputerGuy, for your information.
This is going to be merged this week.

Chocimier commented:
@drake-newell , for your information.
